doc/contribute/bin_blobs.rst

```diff
@@ -0,0 +1,201 @@
+Binary blobs (or blobs for short) are files containing machine code or data in
+proprietary formats, e.g. without corresponding source code released under an
+OSI approved license.
+
+Zephyr supports the distribution and use of binary blobs via its built-in
+mechanisms, with some important caveats, described in the following sections. It
+is important to note that all the information in this section applies only to
+upstream or vanilla Zephyr. There are no limitations whatsoever (except perhaps
+license compatibility) in the support for binary blobs in forks or third-party
+distributions of Zephyr. In fact, Zephyr’s build system supports arbitrary use
+cases related to blobs. This includes linking against libraries, flashing images
+to targets, etc. Users are therefore free to create Zephyr-based downstream
+software which uses binary blobs if they cannot meet the requirements described
+in this page.
+
+Software license
+================
+
+Binary blobs do not typically use OSI approved licenses such as the Apache-2.0
+license. Instead most blobs are distributed under proprietary licenses which
+vary wildly in nature and conditions.
+Licenses will be approved individually by the Technical Steering Committe (TSC)
+upon request of inclusion of a binary blob.
+Blob vendors may impose a click-through or other EULA-like workflow when users
+fetch and install blobs, but this is not required
+
+Hosting
+=======
+
+Blobs must be hosted on the Internet and managed by third-party infrastructure.
+Two potential examples are Git repositories and web servers managed by
+individual hardware vendors.
+
+The Zephyr Project will not host binary blobs in its Git repositories or
+anywhere else.
+
+Distribution
+============
+
+Distribution happens via a west extension command which users
+can run to fetch binary blobs via the Internet. The extension command
+implementation is hosted in the main zephyr repository. This is required to
+give the project overall control over the mechanism used to fetch the blobs.
+
+The blobs themselves must be specified in the :file:`module.yml` files included
+in separate Zephyr :ref:`module repositories <modules>` maintained by their
+respective vendors.  This means that in order to submit a binary blob to the
+upstream Zephyr distribution, a module repository must exist first or be created
+as part of the submission process.
+
+Each blob which may be fetched must be individually identified in the
+corresponding :file:`module.yml` file. A specification for a blob must contain:
+
+- Purpose: an abstract description of the blob itself
+- A reference to vendor-provided documentation
+- The blob’s type, which must be one of the allowed types
+- A checksum for the blob, which the west extension checks after downloading.
+  This is required for reproducibility and to allow bisecting issues as blobs
+  change using Git and west
+- License text applicable to the blob or a reference to such text
+
```

Comment:
Can we add a version information ?
Checksum is ok, but doesn't allow user to perform easy manual identification.